integration_tests/_helpers.py

```diff
@@ -409,15 +445,38 @@ def iter_bytes(self, *args: Any, **kwargs: Any) -> Iterator[bytes]:
         print(self._info.getvalue().decode())
 
 
+def _get_token_from_cookie(cookie: str) -> str:
+    return cookie.split(";", 1)[0].split("=", 1)[1]
+
+
 class _LogTransport(httpx.BaseTransport):
     def __init__(self, transport: httpx.BaseTransport) -> None:
         self._transport = transport
 
     def handle_request(self, request: httpx.Request) -> httpx.Response:
+        assert (port := request.url.port)
+        sequester_tokens = False
+        if request.url.path.endswith("auth/login"):
+            sequester_tokens = DEFAULT_ADMIN_EMAIL in request.content.decode()
+        elif request.url.path.endswith("auth/refresh"):
+            if cookies := request.headers.get("cookie"):
+                sequester_tokens = _DefaultAdminTokenSequestration.intersect(port, cookies)
+        elif (
+            request.url.path.endswith("auth/logout")
+            and (cookies := request.headers.get("cookie"))
+            and _DefaultAdminTokenSequestration.intersect(port, cookies)
+        ):
+            raise RuntimeError("Default admin must not log out during testing.")
         info = BytesIO()
         info.write(f"{'-'*50}\n".encode())
         info.write(f"{datetime.now(timezone.utc).isoformat()}\n".encode())
         response = self._transport.handle_request(request)
+        if (
+            sequester_tokens
+            and response.status_code // 100 == 2
+            and (cookies := response.headers.get("set-cookie"))
+        ):
+            _DefaultAdminTokenSequestration.stash(port, cookies)
```

Comment:
Will we just never logout using the admin tokens?

Reply:
Yes, that is the intent when tests are run concurrently